src/Microsoft.DotNet.Interactive.Js/src/dotnet-interactive/dotnet-interactive-interfaces.ts

```diff
@@ -20,8 +20,24 @@ export interface KernelClient {
     submitCommand(commandType: string, command?: any): Promise<string>;
 }
 
+export interface KernelInvocationContext extends Disposable {
+    subscribeToKernelEvents(observer: KernelEventEnvelopeObserver): DisposableSubscription;
+    complete(commandEnvelope: KernelCommandEnvelope): void;
```

Comment:
this interface should handle KernelCommands and KernelEvents without exposing the envelope types